lib/site_settings/type_supervisor.rb

```diff
@@ -204,7 +204,7 @@ def normalize_input(name, val)
   def validate_value(name, type, val)
     if type == self.class.types[:enum]
       if enum_class(name)
-        raise Discourse::InvalidParameters.new(:value) unless enum_class(name).valid_value?(val)
+        raise Discourse::InvalidParameters.new("Invalid `#{val}` value for `#{name}`") unless enum_class(name).valid_value?(val)
```

Comment:
IMO it would read better like:
```suggestion
        raise Discourse::InvalidParameters.new("Invalid value `#{val}` for `#{name}`") unless enum_class(name).valid_value?(val)
```